packages/python/readme_metrics/GetProjectBaseUrl.py

```diff
@@ -0,0 +1,31 @@
+from functools import lru_cache
+from logging import Logger
+from typing import Dict
+from urllib.parse import urljoin
+
+import base64
+import requests
+
+
+def auth(readme_api_key: str) -> Dict[str, str]:
+    encodedAuth = base64.b64encode(f"{readme_api_key}:".encode("utf8")).decode("utf8")
+    return {"Authorization": f"Basic {encodedAuth}"}
+
+
+@lru_cache(maxsize=None)
```

Comment:
Are we sure that we want to allow the cache to grow without bound? The arguments shouldn't change frequently - could we limit this so as not to blow up memory usage in a client app

Reply:
Agreed, setting bounds would be reasonable. Thoughts on size? We'd need to account for the combined args + return string